app/src/main/java/ke/ac/ku/ledgerly/data/repository/NotificationRepository.kt

```
@@ -0,0 +1,173 @@
+        budgetId: String,
+        category: String,
+        percentageUsed: Double
+    ) {
+        val notification = NotificationEntity(
+            type = "budget_alert",
+            title = "Budget Alert",
+            message = "You've spent ${
+                String.format(
+                    "%.0f",
+                    percentageUsed
+                )
```

detekt (1.23.8)

[warning] String.format(
                    "%.0f",
                    percentageUsed
                ) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)